src/Components/Components/src/Routing/Router.cs

```diff
@@ -147,7 +149,7 @@ private void RefreshRouteTable()
 
         }
 
-        private void Refresh(bool isNavigationIntercepted)
+        internal protected virtual void Refresh(bool isNavigationIntercepted)
```

Comment:
For tests so we can mock out `Refresh`.

Reply:
`internal virtual` is fine, but `protected internal virtual` makes it part of the public API surface. Can this just be `internal virtual`? As long as the test class has InternalsVisibleTo, it should be allowed to override it.

Reply:
Yeah, we can use the `InternalsVisibleTo` attribute here. Updated.